plugins/kubernetes/app/models/kubernetes/build_job_executor.rb

```diff
@@ -0,0 +1,140 @@
+# Execute a job that can build/push Docker images
+# and write progress to the local job output.
+module Kubernetes
+  class BuildJobExecutor
+    WAIT_FOR_JOB = 5.minutes
+    TICK = 3.seconds
+
+    attr_reader :job
+    delegate :id, to: :job
+
+    def initialize(output, job:)
+      @output = output
+      @job = job
+      @registry = {}
+    end
+
+    def execute!(build, project, tag:, push: false, registry:, tag_as_latest: false)
+      job_log = job_name = job_namespace = ""
+      k8s_job = nil
+
+      unless valid_registry_info?(registry)
+        @output.puts "### Registry server should not be empty. Aborting..."
+        return false, job_log
+      end
+      @registry = registry
+
+      # Read the external config path and create a new job config instance
+      contents = File.read(build_job_config_path)
+      k8s_job = Kubernetes::RoleConfigFile.new(contents, build_job_config_path).job
+      fill_job_details(k8s_job, build, project, tag: tag, push: push, tag_as_latest: tag_as_latest)
+
+      if k8s_job
```

Comment:
If there is no job file some succeeding code will likely trigger an exception and be caught in the ```ensure``` block. Is there anything wrong that I am not aware of?

What is your thought on grouping other lines of code into a method? These line look like an one-time glue code between the ```fill_job_details``` and ```create_and_wait_for_job``` to me.